### PR TITLE
PMM-2019 Innodb Compression statistics

### PR DESCRIPTION
**Pull request by adivinho**
added a scraper for innodb cmp, cmpmem

### CODE REVIEW

**Comment by AlekSi** [comment]
LGTM, but let's test it with feature builds first.

**Comment by arvenil** [comment]
@rnovikovP What about this?

File: collector/info_schema_innodb_cmp.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+		if err := informationSchemaInnodbCmpRows.Scan(clientStatScanArgs...); err != nil {
+			return err
+		}
+
+		// Loop over column names, and match to scan data. Unknown columns
+		// will be filled with an untyped metric number. We assume other then
+		// cient, that we'll only get numbers.
+		for idx, columnName := range columnNames[1:] {
+			if metricType, ok := informationSchemaInnodbCmpTypes[columnName]; ok {
+				ch <- prometheus.MustNewConstMetric(metricType.desc, metricType.vtype, float64(clientStatData[idx]), client)
```
Comment (golangci) [inline collector/info_schema_innodb_cmp.go]: unnecessary conversion

File: collector/info_schema_innodb_cmp.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+			} else {
+				// Unknown metric. Report as untyped.
+				desc := prometheus.NewDesc(prometheus.BuildFQName(namespace, informationSchema, fmt.Sprintf("innodb_cmp_%s", strings.ToLower(columnName))), fmt.Sprintf("Unsupported metric from column %s", columnName), []string{"page_size"}, nil)
+				ch <- prometheus.MustNewConstMetric(desc, prometheus.UntypedValue, float64(clientStatData[idx]), client)
```
Comment (golangci) [inline collector/info_schema_innodb_cmp.go]: unnecessary conversion

File: collector/info_schema_innodb_cmpmem.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+		if err := informationSchemaInnodbCmpMemRows.Scan(clientStatScanArgs...); err != nil {
+			return err
+		}
+		// Loop over column names, and match to scan data. Unknown columns
+		// will be filled with an untyped metric number. We assume other then
+		// cient, that we'll only get numbers.
+		for idx, columnName := range columnNames[2:] {
+			if metricType, ok := informationSchemaInnodbCmpMemTypes[columnName]; ok {
+				if columnName == "relocation_time" {
+					ch <- prometheus.MustNewConstMetric(metricType.desc, metricType.vtype, float64(clientStatData[idx]/1000), client, buffer)
```
Comment (golangci) [inline collector/info_schema_innodb_cmpmem.go]: unnecessary conversion

**Pull request merged**